oasislmf/computation/data/dummy_model/generate.py

```diff
@@ -105,42 +172,135 @@ def generate_data(self):
                         np.random.uniform(size=x.shape), 0.0
                     ), 0, triggers
                 )
-                probabilities /= np.sum(probabilities)
+                total_probability = np.sum(probabilities)
+                if (total_probability == 0):
+                    probabilities[0] = 1.0   # First damage bin is always zero-loss
+                else:
+                    probabilities /= total_probability
 
                 for damage_bin, probability in enumerate(probabilities):
                     yield vulnerability + 1, intensity_bin + 1, damage_bin + 1, probability
 
 
 class EventsFile(ModelFile):
+    """
+    Generate random data for Events dummy model file.
+
+    This file lists event IDs to be run.
+
+    Attributes:
+        generate_data: Generate Events dummy model file data.
+    """
+
     def __init__(self, num_events, directory):
+        """
+        Initialise VulnerabilityFile class.
+
+        Args:
+            num_events (int): number of events.
+            directory (str): dummy model file destination.
+        """
         self.num_events = num_events
         self.dtypes = {'event_id': 'i'}
         self.start_stats = None
         self.data_length = num_events
         self.file_name = os.path.join(directory, 'events.bin')
 
     def generate_data(self):
+        """
+        Generate Events dummy model file data.
+
+        Yields:
+            event (int): event ID.
+        """
         return (tuple([event]) for event in range(1, self.num_events + 1))
 
 
-class FootprintFiles(ModelFile):
-    bin_dtypes = OrderedDict([
-        ('areaperil_id', 'i'), ('intensity_bin_id', 'i'), ('probability', 'f')
-    ])
+class FootprintIdxFile(ModelFile):
+    """
+    Generate data for Footprint index dummy model file.
+
+    The binary footprint file footprint.bin requires the index file
+    footprint.idx.
+
+    Attributes:
+        write_file: Write data to Footprint index file in binary format.
+    """
+
+    def __init__(self, directory):
+        """
+        Initialise Footprint index file class.
+
+        Args:
+            directory (str): dummy model file destination.
+        """
+        self.dtypes = OrderedDict([
```

Comment:
https://stackoverflow.com/questions/39980323/are-dictionaries-ordered-in-python-3-6
as of python 3.6, dict are ordered (in 3.7 it is a feature), I would use dict here

Reply:
I agree with Hassan that's better be explicit and use `OrderedDict` if we need the ordering.